### PR TITLE
Add Instance Segmentation workflow wrapper on Object Detection

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s):
Towards KOL-3869

### What change does this PR introduce and why?
Adds an experimental Instance Segmentation workflow which delegates to prebuilt Object Detection workflow. The Object Detection prebulit is capable of handling both bounding boxes and polygons, however it is unintuitive that a Object Detection workflow be used for polygons. This commit adds a thin wrapper on Object Detection to properly classify the workflow as "Instance Segmentation".

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: tests/integration/_experimental/instance_segmentation/test_workflow.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Model = instance_segmentation.Model
+TestCase = instance_segmentation.TestCase
+TestSample = instance_segmentation.TestSample
+TestSuite = instance_segmentation.TestSuite
+EvaluatorConfiguration = instance_segmentation.EvaluatorConfiguration
+InstanceSegmentationEvaluator = instance_segmentation.InstanceSegmentationEvaluator
+
+
+@pytest.mark.metrics
+def test__object_detection__smoke() -> None:
```
Comment (gordonhart) [inline tests/integration/_experimental/instance_segmentation/test_workflow.py]: `test__instance_segmentation__smoke`?

File: kolena/_experimental/instance_segmentation/workflow.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+from kolena.workflow import GroundTruth as BaseGroundTruth
+from kolena.workflow import Inference as BaseInference
+from kolena.workflow.annotation import LabeledBoundingBox
+from kolena.workflow.annotation import LabeledPolygon
+from kolena.workflow.annotation import ScoredLabeledPolygon
+
+
+@dataclass(frozen=True)
+class TestSample(BaseTestSample):
+    """The [`Image`][kolena.workflow.Image] sample type for the pre-built 2D Instance Segmentation workflow."""
```
Comment (gordonhart) [inline kolena/_experimental/instance_segmentation/workflow.py:32]: Unfortunate to have to maintain these docstrings separately from the OD docstrings :/

**Review by gordonhart — APPROVED** [review]
Approving with a few notes. I'd prefer if there were less surface area, but can't see a clear way around it.

File: kolena/_experimental/instance_segmentation/evaluator.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    When a [`EvaluatorConfiguration`][kolena._experimental.instance_segmentation.workflow.EvaluatorConfiguration] is
+    configured to use an F1-Optimal threshold strategy, the evaluator requires that the first test case retrieved for
+    a test suite contains the complete sample set.
+
+    For additional functionality, see the associated [base class documentation][kolena.workflow.evaluator.Evaluator].
+    """
+
+    def __init__(self, configurations: Optional[List[EvaluatorConfiguration]] = None):
+        self.evaluator = ObjectDetectionEvaluator()
```
Comment (marko-polo-cheno) [inline kolena/_experimental/instance_segmentation/evaluator.py:50]: Should we add in optional user configs for single vs multi class evaluation?

File: kolena/_experimental/instance_segmentation/evaluator.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    def compute_test_suite_metrics(
+        self,
+        test_suite: TestSuite,
+        metrics: List[Tuple[TestCase, MetricsTestCase]],
+        configuration: Optional[EvaluatorConfiguration] = None,
+    ) -> Optional[MetricsTestSuite]:
+        return self.evaluator.compute_test_suite_metrics(test_suite, metrics, configuration)
+
+    @staticmethod
+    def _convert_inferences(
```
Comment (marko-polo-cheno) [inline kolena/_experimental/instance_segmentation/evaluator.py:101]: smart! (tiny nit? `_convert_objects`)

**Review by marko-polo-cheno — APPROVED** [review]

File: kolena/_experimental/instance_segmentation/evaluator.py (hunk 3 above)
Comment (consoull) [reply kolena/_experimental/instance_segmentation/evaluator.py:50]: https://github.com/kolenaIO/kolena/pull/353

**Pull request merged**